FV3/atmos_cubed_sphere/tools/fv_restart.F90

```diff
@@ -1422,7 +1423,12 @@ subroutine fv_write_restart(Atm, grids_on_this_pe, timestamp)
     logical, intent(IN) :: grids_on_this_pe(:)
     integer n
 
-    call fv_io_write_restart(Atm, grids_on_this_pe, timestamp)
+    if (all(Atm%flagstruct%do_coarse_graining)) then
+       call fv_io_write_restart_coarse(Atm, grids_on_this_pe, timestamp)
```

Comment:
so we always want to write coarsened restart files when doing coarse graining? Makes sense, why wouldn't you I guess is a better question :).

Reply:
In SHiELD I actually implemented things such that you could specify separately whether you wanted to enable coarse-grained restart files or coarse-grained diagnostics (i.e. I had two flags).  For simplicity I went with one universal flag here, but maybe that wasn't the right choice.  At least for our current use-case, unless we completely abandon the one-step approach toward determining Q1 and Q2, I think we will always do both.  That said, coarse-grained restart files are a fairly niche application that I wouldn't envision a typical user of the model would be interested in.  Coarse-grained diagnostics, on the other hand, would likely have a broader appeal.

Just writing this down seems to re-convince me that having two flags is probably better than one.  I think I'll make that change.